shared/src/k8s/mod.rs

```diff
@@ -621,6 +622,58 @@ impl KubeInterface for KubeImpl {
     }
 }
 
+/// This deletes an Instance unless it has already been deleted by another node
+/// or fails after multiple retries.
+pub async fn try_delete_instance(
+    kube_interface: &dyn KubeInterface,
+    instance_name: &str,
+    instance_namespace: &str,
+) -> Result<(), Box<dyn std::error::Error + Send + Sync + 'static>> {
+    for x in 0..MAX_INSTANCE_UPDATE_TRIES {
+        match kube_interface
+            .delete_instance(instance_name, &instance_namespace)
+            .await
+        {
+            Ok(()) => {
+                log::trace!("try_delete_instance - deleted Instance {}", instance_name);
+                break;
+            }
+            Err(e) => {
+                // Check if already was deleted else return error
+                match kube_interface
+                    .find_instance(&instance_name, &instance_namespace)
+                    .await
+                {
+                    Err(kube::Error::Api(ae)) => {
+                        if ae.code == ERROR_NOT_FOUND {
+                            log::trace!(
+                                "try_delete_instance - discovered Instance {} already deleted",
+                                instance_name
+                            );
+                            break;
+                        }
+                        log::error!("try_delete_instance - when looking up Instance {}, got kube API error: {:?}", instance_name, ae);
+                    }
+                    Err(e) => {
+                        log::error!("try_delete_instance - when looking up Instance {}, got kube error: {:?}. {} retries left.", instance_name, e, MAX_INSTANCE_UPDATE_TRIES - x - 1);
+                    }
+                    Ok(_) => {
+                        log::error!(
+                            "try_delete_instance - tried to delete Instance {} but still exists. {} retries left.",
+                            instance_name, MAX_INSTANCE_UPDATE_TRIES - x - 1
+                        );
+                    }
+                }
+                if x == MAX_INSTANCE_UPDATE_TRIES - 1 {
+                    return Err(e);
+                }
+            }
+        }
+        random_delay().await;
```

Comment:
What is the cap on the random_delay?

Reply:
up to 200 ms. It is noted in [in-line doc of the function](https://github.com/kate-goldenring/akri/blob/72f13fac7843bbd87a6924f6809cb66e1cd9894f/shared/src/akri/mod.rs#L26). should it also be noted elsewhere?